### PR TITLE
Update Scheduler

### LINKED ISSUE #487: Scheduler raises a warning
UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
  warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)
**why this warnings?**

### PR DESCRIPTION
**Pull request by CCInc**
Trying to fix  #487. The change steps the scheduler however many times. 

Note that the scheduler that updates on samples changes the learning rate before the epoch scheduler. The epoch scheduler has to update after the first step of the new epoch, while the sample scheduler updates after the last step of the current epoch. This was avoided before by setting `_num_samples` to `-1`, but this no longer works. 

Let me know any thoughts/suggestions if there's a better way to implement this!

### CODE REVIEW

**Review by nicolas-chaulet — APPROVED** [review]
Looks great to me, thanks a lot for that! It was a long overdue task 🙏
